src/safari-app-extension/AlpheiosReadingTools/AppDelegate.swift

```diff
@@ -371,68 +353,51 @@ class AppDelegate: NSObject, NSApplicationDelegate {
         os_log("Logout has been initiated", log: OSLog.sAlpheios, type: .info)
         #endif
         
-        self.authText.stringValue = "You are logged out"
+        self.isAuthenticated = false
+        self.userNickname = nil
+        self.updateAuthUI()
         
-        SFSafariApplication.dispatchMessage(withName: "UserLogout", toExtensionWithIdentifier: "net.alpheios.safari.ext", userInfo: nil, completionHandler: nil)
+        SFSafariApplication.dispatchMessage(withName: "UserLogout", toExtensionWithIdentifier: "net.alpheios.safari.ext", userInfo: nil, completionHandler: { (error) -> Void in
+            if (error == nil) {
+                os_log("Logout message to the Safari App Extension has been dispatched successfully", log: OSLog.sAlpheios, type: .info)
+            } else {
+                os_log("Logout message to the Safari App Extension has failed: %s", log: OSLog.sAlpheios, type: .error, error!.localizedDescription)
```

Comment:
is there any value in showing an error to the user in this case?

Reply:
This message will be printed to the logs if we cannot send a message to the Safari App Extension. I was thinking that it will be the case when something is very wrong in the system and we'd better have a record about it.

Reply:
yes I understand the reason for the log message. I was just wondering if we ought to try to alert the user that something went wrong.

Reply:
Oh, sorry for misreading it. I think it's a good idea. We can show an error message in the containing app's UI. Will update the code.

Reply:
Thanks!